File: library/src/main/java/io/vithor/yamvpframework/TextUtils.java
package io.vithor.yamvpframework;

import android.text.Editable;

/**
 * Created by Hazer on 1/5/16.
 */
public class TextUtils {
    public static boolean isEmpty(String... strings) {
        for (String str : strings) {
            if (str.isEmpty())
            return true;
        }
        return false;
    }

    public static String onlyNumbers(Editable text) {
        return text.toString().replaceAll("[^\\d]", "");
    }

    public static String clearText(Editable text) {
        return text.toString().replaceAll("[^a-zA-Z0-9]", "");
    }
}

File: ui/src/main/java/io/vithor/yamvpframework/ui/TextUtils.kt
package io.vithor.yamvpframework

import android.text.Editable

/**
 * Created by Vithorio Polten on 1/5/16.
 */
object TextUtils {
    fun isEmpty(vararg strings: String?): Boolean {
        return strings.any { it.isNullOrEmpty() }
    }

    fun isBlank(vararg strings: String?): Boolean {
        return strings.any { it.isNullOrBlank() }
    }

    fun onlyNumbers(text: Editable): String {
        return text.toString().replace("[^\\d]".toRegex(), "")
    }

    fun clearText(text: Editable): String {
        return text.toString().replace("[^a-zA-Z0-9]".toRegex(), "")
    }
}
